OpenRA.Game/HttpExtension.cs

```diff
@@ -0,0 +1,61 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenRA
+{
+	public delegate void OnProgress(long total, long totalRead, int progressPercentage);
+
+	public static class HttpExtension
+	{
+		public static async Task ReadAsStreamWithProgress(this HttpResponseMessage response, Stream outputStream, OnProgress onProgress, CancellationToken token)
+		{
+			var total = response.Content.Headers.ContentLength ?? -1;
+			total = total > 0 ? total : -1;
+
+			var canReportProgress = total != -1;
```

Comment:
The addition of `total = total > 0 ? total : -1;` above makes me think you can do this instead:
```suggestion
			var canReportProgress = total > 0;
```
(and remove the extra assignment above, ofc).